File: neo4j-php-java-bridge/src/main/java/com/petroquick/neo4j/bench/SimpleTest.java
package com.petroquick.neo4j.bench;

import org.neo4j.graphdb.*;
import org.neo4j.index.IndexService;
import org.neo4j.kernel.EmbeddedGraphDatabase;
import org.neo4j.kernel.impl.batchinsert.BatchInserter;
import org.neo4j.kernel.impl.batchinsert.BatchInserterImpl;

import java.io.File;
import java.util.HashMap;
import java.util.Map;

/**
 * @author Philip Rud
 */
public class SimpleTest {
    public static final String NEO4J_DB = "./db-j";

    public static EmbeddedGraphDatabase neo;
    public static IndexService indexer;
    public static BatchInserter inserter;

    public static Long prevTime = null;

    public static void printTime(String message) {
        long currentTime = System.nanoTime();

        if (prevTime == null)
            prevTime = currentTime;
                                        
        System.out.println(message + "\ntime = " + (currentTime - prevTime) / 1000000000.0 + " sec\n");
        prevTime = currentTime;
    }

    public static void clearDir(String path, boolean deleteSelf) {
        File file = new File(path);
        if (!file.exists())
            throw new RuntimeException("path not found");    

        if (file.isDirectory())
            for (String child: file.list())
                clearDir(path + "/" + child, true);

        if (deleteSelf)
            if (!file.delete())
                throw new RuntimeException("couldn't delete some files");
    }

    public static void createChildren(long parent, int childrenPerLevel, int levelsMore) {
        for (int i = 0; i < childrenPerLevel; i++) {
            Map<String, Object> properties = new HashMap<String, Object>();
            properties.put("i", i);
            properties.put("level", levelsMore + 1);
            long node = inserter.createNode(properties);

            inserter.createRelationship(parent, node, DynamicRelationshipType.withName("PARENT"), null);

            if (levelsMore > 0)
                createChildren(node, childrenPerLevel, levelsMore - 1);
        }
    }

    public static void main(String[] args) {
        clearDir(NEO4J_DB, false);
        printTime("initializing...");

        neo = new EmbeddedGraphDatabase(NEO4J_DB);
        neo.shutdown();
        printTime("clean db created");

        inserter = new BatchInserterImpl(NEO4J_DB);
        createChildren(inserter.getReferenceNode(), 100, 1);
        inserter.shutdown();
        printTime("nodes created");

        for (int i = 0; i < 10; i++) {
            neo = new EmbeddedGraphDatabase(NEO4J_DB);

            System.out.println("traversing through all nodes and counting depthSum...");
            Transaction tx = neo.beginTx();
            Traverser traverser = neo.getReferenceNode().traverse(
                    Traverser.Order.BREADTH_FIRST,
                    StopEvaluator.END_OF_GRAPH,
                    ReturnableEvaluator.ALL_BUT_START_NODE,
                    DynamicRelationshipType.withName("PARENT"),
                    Direction.OUTGOING
            );

            int depthSum = 0;
            for (Node node: traverser)
                depthSum += (Integer) node.getProperty("level");

            tx.finish();
            tx.success();
            System.out.println("depthSum = " + depthSum);
            printTime("done traversing");

            neo.shutdown();
        }
        
        System.out.println("neo has been shut down");
    }

}

File: neo4j-php-java-bridge/src/main/webapp/php/PetroQuick/Neo4J/Bench/SimpleTest.php
<?php

require_once(PetroQuick_Neo4JGateway_Config::GATEWAY_URL);

class PetroQuick_Neo4J_Bench_SimpleTest {
    public static $NEO4J_DB;

    public static $neo;
    public static $indexer;
    public static $inserter;

    public static $prevTime = null;

    public static function printTime($message) {
        microtime(true);

        $currentTime = microtime(true);

        if (!isset(self::$prevTime))
            self::$prevTime = $currentTime;

        echo "\n$message" . "\ntime = " . sprintf("%F", ($currentTime - self::$prevTime)) . " sec\n";
        self::$prevTime = $currentTime;
    }

    public static function clearDir($path, $deleteSelf) {
        if (is_dir($path)) {
            $children = glob(rtrim($path, '/') . '/*');
            foreach ($children as $index => $child)
                self::clearDir($child, true);
        }

        if ($deleteSelf) {
            if (is_dir($path))
                rmdir($path);
            else
                unlink($path);
        }
    }

    public static function createChildren($parent, $childrenPerLevel, $levelsMore) {
        for ($i = 0; $i < $childrenPerLevel; $i++) {
            $properties = new java("java.util.HashMap");
            $properties->put("i", $i);
            $properties->put("level", $levelsMore + 1);
            $node = self::$inserter->createNode($properties);

            self::$inserter->createRelationship($parent, $node, java("org.neo4j.graphdb.DynamicRelationshipType")->withName("PARENT"), null);

            if ($levelsMore > 0)
                self::createChildren($node, $childrenPerLevel, $levelsMore - 1);
        }
    }

    public static function main() {
        echo "<pre>";

        self::printTime("initializing...");

        self::clearDir(self::$NEO4J_DB, false);

        self::$neo = new java("org.neo4j.kernel.EmbeddedGraphDatabase", self::$NEO4J_DB);
        self::$neo->shutdown();
        self::printTime("clean db created");

        self::$inserter = new java("org.neo4j.kernel.impl.batchinsert.BatchInserterImpl", self::$NEO4J_DB);
        self::createChildren(self::$inserter->getReferenceNode(), 10, 1);
        self::$inserter->shutdown();
        self::printTime("nodes created");

        self::$neo = new java("org.neo4j.kernel.EmbeddedGraphDatabase", self::$NEO4J_DB);

        for ($i = 0; $i < 10; $i++) {
            echo "\ntraversing through all nodes and counting depthSum...";
            $tx = self::$neo->beginTx();
            $traverser = self::$neo->getReferenceNode()->traverse(
                java('org.neo4j.graphdb.Traverser$Order')->BREADTH_FIRST,
                java('org.neo4j.graphdb.StopEvaluator')->END_OF_GRAPH,
                java('org.neo4j.graphdb.ReturnableEvaluator')->ALL_BUT_START_NODE,
                java('org.neo4j.graphdb.DynamicRelationshipType')->withName('PARENT'),
                java('org.neo4j.graphdb.Direction')->OUTGOING
            );

            $depthSum = 0;
            $nodes = $traverser->iterator();
            while (java_values($nodes->hasNext())) {
                $node = $nodes->next();
                $depthSum += intval(java_values($node->getProperty("level")));
            }

            $tx->finish();
            $tx->success();
            echo "\ndepthSum = $depthSum";
            self::printTime("done traversing");
        }

        self::$neo->shutdown();
                              
        echo "\nneo has been shut down";
        echo "</pre>";
    }


}